extensions/ql-vscode/src/run-queries.ts

```diff
@@ -635,8 +635,8 @@ export async function compileAndRunQueryAgainstDatabase(
         formattedMessages.push(formatted);
         qs.logger.log(formatted);
       }
-      if (quickEval && formattedMessages.length <= 3) {
-        showAndLogErrorMessage('Quick evaluation compilation failed: \n' + formattedMessages.join('\n'));
+      if (quickEval && formattedMessages.length <= 2) {
+        showAndLogErrorMessage('Quick evaluation compilation failed: ' + formattedMessages.join('\n'));
```

Comment:
This section seems strange to me. Why `<= 2`? This isn't a problem with your code since it was `<= 3` and neither magic number is clear to me. If you understand this, can you add a comment here? If not, then can you add a `TODO` so that we can hopefully fix later.

Reply:
My understanding is that this is simply a number of compilation errors that it would be reasonable to show in a pop-up, and if there's more errors than that we instead show the generic error in `compilationFailedErrorTail` which says to look at the full output in the log. I think changing it from `3` to `2` is reasonable if we are limiting other log messages to two lines - three seems like a lot of compilation messages to try and show in a pop-up anyways.